http.go

```diff
@@ -256,51 +259,120 @@ func Get(w http.ResponseWriter, req *http.Request, store mdata.Store, metricInde
 
 	reqs := make([]Req, 0)
 	for _, target := range targets {
-		var consolidateBy string
-		id := target
-		// yes, i am aware of the arguably grossness of the below.
-		// however, it is solid based on the documented allowed input format.
-		// once we need to support several functions, we can implement
-		// a proper expression parser
-		if strings.HasPrefix(target, "consolidateBy(") {
-			t := target
-			if t[len(t)-2:] != "')" || (!strings.Contains(t, ",'") && !strings.Contains(t, ", '")) || strings.Count(t, "'") != 2 {
-				http.Error(w, "target parse error", http.StatusBadRequest)
-				return
-			}
-			consolidateBy = target[strings.Index(target, "'")+1 : strings.LastIndex(target, "'")]
-			id = target[strings.Index(target, "(")+1 : strings.LastIndex(target, ",")]
+		id, consolidateBy, err := parseTarget(target)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
 		}
 
+		type locatedDef struct {
+			def schema.MetricDefinition
+			loc string
+		}
+
+		locatedDefs := make(map[string]locatedDef)
+
 		if legacy {
 			nodes, err := metricIndex.Find(org, id)
 			if err != nil {
 				http.Error(w, err.Error(), http.StatusBadRequest)
 				return
 			}
-			if len(nodes) == 0 {
-				http.Error(w, errMetricNotFound.Error(), http.StatusBadRequest)
-				return
-			}
 			for _, node := range nodes {
 				for _, def := range node.Defs {
-					consolidator, err := consolidation.GetConsolidator(&def, consolidateBy)
-					if err != nil {
-						http.Error(w, err.Error(), http.StatusBadRequest)
-						return
+					locatedDefs[def.Id] = locatedDef{def, "local"}
+				}
+			}
+
+			for _, inst := range otherNodes {
+				res, err := http.Get(inst)
+				http.PostForm("http://%s/index/find", url.Values{"target": targets})
+				if err != nil {
+					http.Error(w, err.Error(), http.StatusInternalServerError)
+				}
+				if res.StatusCode != 200 {
+					// if the remote returned interval server error, or bad request, or whatever, we want to relay that as-is to the user.
+					msg, _ := ioutil.ReadAll(res.Body)
+					http.Error(w, string(msg), res.StatusCode)
+				}
+				defer res.Body.Close()
+				buf, _ := ioutil.ReadAll(res.Body)
+				var n idx.Node
+				for len(buf) != 0 {
+					buf, err = n.UnmarshalMsg(buf)
+					// different nodes may have overlapping data in their index.
+					// maybe because they loaded the entire index from a persistent store,
+					// or they used to receive a certain shard.
+					// so we need to select the node that has most recently seen each given metricDef.
+					for _, def := range n.Defs {
+						cur, ok := locatedDefs[def.Id]
+						if ok && cur.def.LastUpdate > def.LastUpdate {
```

Comment:
this should be ">=" so that if the local node and another have the same metricDef, the local node is preferred.